xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-repositories/xwiki-commons-extension-repository-http/src/main/java/org/xwiki/extension/repository/http/internal/DefaultHttpClientFactory.java

```diff
@@ -57,6 +58,35 @@ public CloseableHttpClient createClient(String user, String password)
         // Pre-configure with everything configured at JVM level
         httpClientBuilder.useSystemProperties();
 
+        HttpHost proxy = null;
+        CredentialsProvider proxyCredentialsProvider = null;
+        String proxyHost = System.getProperty("http.proxyHost");
+        String proxyPort = System.getProperty("http.proxyPort");
+        if ((proxyHost != null) && !proxyHost.isEmpty()) {
+            int port = 3128;
+            if (proxyPort != null && !proxyPort.isEmpty()) {
+                port = Integer.parseInt(proxyPort);
+            }
+
+            proxy =  new HttpHost(proxyHost, port);
+            httpClientBuilder.setProxy(proxy);
```

Comment:
An much simpler alternative to all that might be to just set a default java.net.Authenticator which return whatever is indicated in http.proxyUser/Pass if called with RequestorType.PROXY.